accepted/2023/preview-apis/preview-apis.md

```diff
@@ -0,0 +1,279 @@
+# Experimental APIs
+
+**Owner** [Immo Landwerth](https://github.com/terrjobst)
+
+In .NET 6, we've added a way to [ship preview features][preview-features] in
+otherwise stable releases. It works by tagging APIs (and language features) as
+preview features and requiring the user's project to explicitly opt-into using
+them via the `<EnablePreviewFeatures>` property.
+
+The feature was primarily designed to accommodate platform-level preview
+features, that is for features that span runtime, library, and language, such as
+generic math. However, we also believed that the feature would work for
+libraries that want to offer preview functionality in otherwise stable NuGet
+packages.
+
+Since then, we've seen requests from partners who have tried using it and hit
+limitations that come from guardrails we put into place due to the complex
+nature of platform-level preview features.
+
+The goal of this feature is to find a design that works for both, platform-level
+preview features and library-level preview features.
+
+Let's first define these two terms:
+
+* **platform-level preview feature**. These are features that rely on preview
+  behaviors across runtime, core library, tooling, and languages. The nature of
+  these features is that they are in support of each other. For example, generic
+  math requires new runtime behaviors for interfaces, which the language has to
+  support, and of course new interfaces and methods in core library that
+  leverages them. If you want to use generic math, you need a combination of
+  runtime, framework, and language that were designed to work in conjunction.
+
+* **library-level preview feature**. These are library APIs that don't require
```

Comment:
Might also interesting to scope in source generators to this.

Reply:
Tell me more about the source generator case. How are they different from manually written code?

Reply:
Could require experimental compiler features, are also not (always) addressed with API in reference assemblies.  Probably there are other wrinkles.  I didn't have any major problems in mind, but felt they were different enough that we should walk through what an experimental source generator would look like.  I can also imagine we have cases of these that we'd want to represent as experimental.